packages/integrations/src/download-client/nzbget/nzbget-integration.ts

```diff
@@ -64,39 +61,54 @@ export class NzbGetIntegration extends DownloadClientIntegration {
   }
 
   public async pauseQueueAsync() {
-    await this.getClient().pausedownload();
+    await this.nzbGetApiCallAsync("pausedownload");
   }
 
   public async pauseItemAsync({ id }: DownloadClientItem): Promise<void> {
-    await this.getClient().editqueue("GroupPause", "", [Number(id)]);
+    await this.nzbGetApiCallAsync("editqueue", "GroupPause", "", [Number(id)]);
   }
 
   public async resumeQueueAsync() {
-    await this.getClient().resumedownload();
+    await this.nzbGetApiCallAsync("resumedownload");
   }
 
   public async resumeItemAsync({ id }: DownloadClientItem): Promise<void> {
-    await this.getClient().editqueue("GroupResume", "", [Number(id)]);
+    await this.nzbGetApiCallAsync("editqueue", "GroupResume", "", [Number(id)]);
   }
 
   public async deleteItemAsync({ id, progress }: DownloadClientItem, fromDisk: boolean): Promise<void> {
-    const client = this.getClient();
     if (fromDisk) {
-      const filesIds = (await client.listfiles(0, 0, Number(id))).map((value) => value.ID);
-      await this.getClient().editqueue("FileDelete", "", filesIds);
+      const filesIds = ((await this.nzbGetApiCallAsync("listfiles", 0, 0, Number(id))) as { ID: number }[]).map(
+        (file) => file.ID,
+      );
+      await this.nzbGetApiCallAsync("editqueue", "FileDelete", "", filesIds);
     }
-    if (progress !== 1) {
-      await client.editqueue("GroupFinalDelete", "", [Number(id)]);
+    if (progress === 1) {
+      await this.nzbGetApiCallAsync("editqueue", "GroupFinalDelete", "", [Number(id)]);
     } else {
-      await client.editqueue("HistoryFinalDelete", "", [Number(id)]);
+      await this.nzbGetApiCallAsync("editqueue", "HistoryFinalDelete", "", [Number(id)]);
     }
   }
 
-  private getClient() {
+  private async nzbGetApiCallAsync(method: string, ...params: unknown[]): Promise<unknown> {
     const url = new URL(this.integration.url);
     url.pathname += `${this.getSecretValue("username")}:${this.getSecretValue("password")}`;
```

Comment:
Is there a better way to pass username / password?

Reply:
Unfortunately no...